### PR TITLE
Remove some deprecated pieces of sheerlike

### PR DESCRIPTION
**Pull request by chosak**
This change removes the last remaining Sheerlike site ("owning-a-home"), along with a few bits of the `sheerlike` app that can be cleanly removed. There are still other pieces of `sheerlike` that are still in use.

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [X] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: cfgov/cfgov/settings/base.py [hunk 1]
```diff
@@ -465,14 +465,6 @@
 LOGIN_REDIRECT_URL = '/login/welcome/'
 LOGIN_URL = "/login/"
 
-
```
Comment (sebworks) [inline cfgov/cfgov/settings/base.py]: We still need this code.

Comment (chosak) [reply cfgov/cfgov/settings/base.py]: Where does this code still get used?

Comment (sebworks) [reply cfgov/cfgov/settings/base.py]: for the rate checker; that flag hasn't been flipped yet.

Comment (chosak) [reply cfgov/cfgov/settings/base.py]: Right, that path still [uses](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/cfgov/urls.py#L86) `SheerTemplateView`, but that doesn't seem to depend on the Sheer Site at all. All that's really doing is modifying how/where the template file gets loaded from.

**Comment by chosak** [comment]
Marking on hold pending ratechecker migration.

File: cfgov/cfgov/settings/base.py (hunk 1 above)
Comment (anselmbradford) [reply cfgov/cfgov/settings/base.py]: The flag has now been flipped in production.

**Pull request merged**